pkg/util/ciutil/gitlab.go

```diff
@@ -27,13 +27,13 @@ type gitlabCI struct {
 // See https://docs.gitlab.com/ee/ci/variables/.
 func (gl gitlabCI) DetectVars() Vars {
 	v := Vars{Name: gl.Name}
-	v.BuildID = os.Getenv("CI_JOB_ID")
+	v.BuildID = os.Getenv("CI_PIPELINE_IID")
```

Comment:
Although the previous env var had a valid value, it was a unique ID within GitLab's environment and so it looks odd when rendering it in the UI. The `CI_PIPELINE_IID` is an instance ID within the context of the current GitLab project and is expected to be a more sane number and not something like `406044007`.